src/main/java/com/ddudu/user/repository/UserRepositoryImpl.java

```diff
@@ -20,6 +23,19 @@ public UserRepositoryImpl(EntityManager em) {
   }
 
   @Override
+  public List<User> findAllByKeywordAndSearchType(String keyword, UserSearchType userSearchType) {
+    JPAQuery<User> query = jpaQueryFactory
+        .selectFrom(user);
+
+    switch (userSearchType) {
+      case EMAIL -> query.where(user.email.address.eq(keyword));
+      case NICKNAME -> query.where(user.nickname.eq(keyword));
+      case OPTIONAL_USERNAME -> query.where(user.optionalUsername.eq(keyword));
+    }
+
+    return query.fetch();
+  }
```

Comment:
```suggestion
@Override
public List<User> findAllByKeywordAndSearchType(String keyword, UserSearchType userSearchType) {
    BooleanBuilder whereClause = new BooleanBuilder();

    switch (userSearchType) {
        case EMAIL -> whereClause.and(user.email.address.eq(keyword));
        case NICKNAME -> whereClause.and(user.nickname.eq(keyword));
        case OPTIONAL_USERNAME -> whereClause.and(user.optionalUsername.eq(keyword));
    }

    return jpaQueryFactory
            .selectFrom(user)
            .where(whereClause)
            .fetch();
}
```

혹시 이렇게 리팩토링하는 건 어떻게 생각하시나요? 개인적으로 쿼리문은 모여있어야 가독성이 좋다고 생각이 들어서요! 물론 의진님의 취향대로 그대로 진행하셔도 됩니다 💯

Reply:
좋은 것 같습니당! 적용하겠습니다:)